contracts/Splitter.sol

```diff
@@ -23,25 +31,28 @@ contract Splitter {
 
   function split() external payable onlyAlice() {
     require(msg.value > 0, "Must split more than zero ether!");
-    require(msg.value % 2 == 0, "The ether to be splitted must be even!");
+    require(msg.value.mod(2) == 0, "The ether to be splitted must be even!");
 
-    uint256 half = msg.value / 2;
-    bobBalance += half;
-    carolBalance += half;
+    uint256 half = msg.value.div(2);
+    bobBalance = bobBalance.add(half);
+    carolBalance = carolBalance.add(half);
+    emit LogAliceSplitted(msg.value);
   }
 
   function withdraw() external {
     require(msg.sender == bob || msg.sender == carol, "Only Bob or Carol can withdraw!");
     if (msg.sender == bob) {
-      require(bobBalance > 0, "Bob's balance must be grater than zero!");
       uint256 tempBobBalance = bobBalance;
+      require(tempBobBalance > 0, "Bob's balance must be grater than zero!");
       bobBalance = 0;
       msg.sender.transfer(tempBobBalance);
+      emit LogBobWithdrawn(tempBobBalance);
```

Comment:
If the recipient emits an event on reception, in which order will these 2 events be?

Reply:
The reception event will emit first since the transfer runs first, is it right?

Reply:
Yep. Is it an order you like? `LogReceived`, then `LogWithdrawn`.

Reply:
you are right, should be the other way around, thanks